src/components/Layout/Header/Header.tsx

```diff
@@ -98,19 +100,31 @@ export const Header = memo(() => {
 
   const currentWalletId = useSelector(selectWalletId)
   const walletAccountIds = useSelector(selectWalletAccountIds)
-  const hasUtxoAccountIds = useMemo(
-    () => walletAccountIds.some(accountId => isUtxoAccountId(accountId)),
-    [walletAccountIds],
-  )
 
   useEffect(() => {
     const isMetaMaskMultichainWallet = wallet instanceof MetaMaskShapeShiftMultiChainHDWallet
     if (!(currentWalletId && isMetaMaskMultichainWallet && isSnapInstalled === false)) return
 
-    // We have just detected that the user doesn't have the snap installed currently
-    // We need to check whether or not the user had previous non-EVM AccountIds and clear those
-    if (hasUtxoAccountIds) appDispatch(portfolio.actions.clearWalletMetadata(currentWalletId))
-  }, [appDispatch, currentWalletId, hasUtxoAccountIds, isSnapInstalled, wallet, walletAccountIds])
+    const nonEvmAccountIds = walletAccountIds.filter(
+      accountId => !isEvmChainId(fromAccountId(accountId).chainId),
+    )
+
+    // We have just detected that the user doesn't have the snap installed currently, so disable all non-evm accounts
+    nonEvmAccountIds.forEach(accountId =>
+      appDispatch(portfolio.actions.disableAccountId(accountId)),
+    )
```

Comment:
This diff sounds off - this bit shouldn't change as far as this fix is concerned?
i.e we definitely should nuke accounts off portfolio in this case, not disable them, which has a whole different meaning, means AccountIds still exist in the store despite not being part of the user's wallet anymore.

Barring fluff AccountIds concerns, there is a risk of these being disabled while consequently re-installing the snap (assumedly edge-case from a user perspective, but goes to show this is probably not what we want)